main.tf

```diff
@@ -78,15 +78,15 @@ data "aws_iam_policy_document" "logs_policy_doc" {
 # Create the IAM role for the Lambda instance.
 resource "aws_iam_role" "main" {
   name               = "lambda-${local.full_name}"
-  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
 }
 
 # Attach the logging policy to the above IAM role.
 resource "aws_iam_role_policy" "main" {
   name = "lambda-${local.full_name}"
-  role = "${aws_iam_role.main.id}"
+  role = aws_iam_role.main.id
 
-  policy = "${data.aws_iam_policy_document.logs_policy_doc.json}"
+  policy = data.aws_iam_policy_document.logs_policy_doc.json
 }
 
 # This code verifies that the count of policy ARNs matches the actual
```

Comment:
Do we still need this workaround? I'm not sure I understand the particular limitation that Terraform has/had here and if that got fixt in TF12?

Reply:
I've been thinking about potentially doing this in 2 phases 
1) to make our modules syntactically work with 0.12 
2) refactor the modules to more closely align with the 0.12 feature set.

That being said I'm open to other opinions.

Reply:
I *think* this was the Terraform issue that was forcing this workaround: https://github.com/hashicorp/terraform/issues/14677. This is supposedly fixed in 0.12 from the sound of it. I tend to agree with @dynamike that a two-stage process works pretty well. I also wanted to try and get this sorted out ASAP under the assumption we can make it prettier later.

Reply:
in which case, I'll approve.